compiler/src/main/java/org/qbicc/type/CompoundType.java

```diff
@@ -281,7 +281,7 @@ public CompoundType build() {
                 throw new IllegalStateException("CompoundType has no members");
             }
             if (completeType == null) {
-                int size = offset;  // Offset points to the end of the structure
+                int size = (offset + (overallAlign - 1)) & -overallAlign;  // Offset points to the end of the structure, align the size to overall alignment
```

Comment:
I had to look up the C structure sizing and alignment rules to convince myself that this was correct (and it is).  Including the tail padding in the size helps ensure optimal alignment for arrays so they avoid unaligned accesses on oddly sized structs.